deploy/setup-project.sh

```diff
@@ -12,24 +12,24 @@ IAM_NAME="${GCEPD_SA_NAME}@${PROJECT}.iam.gserviceaccount.com"
 # Create or Update Custom Role
 if gcloud iam roles describe gcp_compute_persistent_disk_csi_driver_custom_role --project "${PROJECT}";
 then
-  yes | gcloud iam roles update gcp_compute_persistent_disk_csi_driver_custom_role \
-          --project "${PROJECT}"                                                   \
+  gcloud iam roles update gcp_compute_persistent_disk_csi_driver_custom_role --quiet \
+          --project "${PROJECT}"                                                     \
           --file "${PKGDIR}/deploy/gcp-compute-persistent-disk-csi-driver-custom-role.yaml"
 else
-  gcloud iam roles create gcp_compute_persistent_disk_csi_driver_custom_role \
-    --project "${PROJECT}"                                                   \
+  gcloud iam roles create gcp_compute_persistent_disk_csi_driver_custom_role --quiet \
+    --project "${PROJECT}"                                                           \
     --file "${PKGDIR}/deploy/gcp-compute-persistent-disk-csi-driver-custom-role.yaml"
 fi
 
 # Delete Service Account Key
 if [ -f $SA_FILE ]; then
   rm "$SA_FILE"
 fi
-# Delete Bindings
-for role in ${BIND_ROLES}
-do
-  gcloud projects remove-iam-policy-binding "${PROJECT}" --member serviceAccount:"${IAM_NAME}" --role $role --quiet || true
-done
+# Delete ALL EXISTING Bindings
+gcloud projects get-iam-policy "${PROJECT}" --format json > "${PKGDIR}/deploy/iam.json"
+sed -i "/serviceAccount:${IAM_NAME}/d" "${PKGDIR}/deploy/iam.json"
+gcloud projects set-iam-policy "${PROJECT}" "${PKGDIR}/deploy/iam.json"
```

Comment:
hmm although in general I find this kind of scary because it sets all the rolebindings in your entire project to the .json, not just this particular service account.  If someone is also modifying the iam policies at the same time, this could override their changes.

Reply:
theres an etag that is used for consistency. `gcloud projects set-iam-policy` will fail if the etag is out of date. I haven't written in any failure handling logic. It shouldn't be too difficult to just run the script again.